### DISCUSSION TITLE
OT: [haskell][c2hs][Language.C] The symbol `__uint128_t' does not fit here

### DISCUSSION THREAD

**Discussion by mamat1123** [jedisct1/libsodium#1264] [Q&A]
ghc version 8.10.7
xcode version 14.1

```
 > configure
libsodium              > Configuring libsodium-1.0.18.2...
libsodium              > build
libsodium              > Preprocessing library for libsodium-1.0.18.2..
libsodium              > c2hs: C header contains errors:
libsodium              >
libsodium              > /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/mach/arm/_structs.h:498: (column 2) [ERROR]  >>> Syntax error !
libsodium              >   The symbol `__uint128_t' does not fit here.
libsodium              >

```

**Comment by jedisct1** [comment]
Where does that `configure` script come from?

The output doesn't match at all the output from the `configure` script shipped with this project.

**Comment by jedisct1** [comment]
The first result of a Google search for "The symbol `__uint128_t' does not fit here" is:

https://discourse.haskell.org/t/problem-with-language-c-on-arm-mac/3841

that seems to be quite relevant.

https://github.com/visq/language-c/pull/89

Looks like it's an old issue in `Language.C` (no idea what that is, but it feels like what you are using) that has been fixed. So, you should try to update that.

**Comment by CordiaMcmartini** [comment]
The error message suggests that there is a syntax error in the C header file /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/mach/arm/_structs.h. Specifically, it appears to be complaining about the symbol __uint128_t.

One possible solution would be to try upgrading your version of libsodium, as this may include a fix for this issue. Alternatively, you could try modifying the C header file to fix the syntax error. However, this can be tricky, as modifying system header files can have unexpected consequences.

Another option would be to try using a different compiler or development environment to build your project. For example, you could try using Clang instead of GHC, or try building on a different operating system. This may help to avoid issues with the header file.
